bgapi/module.py

```diff
@@ -216,10 +217,18 @@ def procedure_call(self, procedure_type, timeout, throwError=True):
             
             handle = ProcedureCallHandle()
             self._handles[procedure_type] = handle
+
+            # Ensure that the lib does not exceed connection interval.
+            earliest_interval_ts = self.ioTimestamps[procedure_type][0]
+            interval = self.get_procedure_call_interval()
+            while (time.time() - earliest_interval_ts) < interval:
+                time.sleep(max(0.01, time.time() - earliest_interval_ts))
```

Comment:
Have you tested this much? I don't think this strategy will work.  Even though most devices use a fixed number of packets per interval, the remote device can start rejecting at any point in the received packets according to the specification.  Most of the devices I use allow 6 packets per interval, so at a minimum the maxlen would need to a be configurable parameter.  In my experience, when I overload the remote device I get an "Out Of Memory" response to the write without response command.  I tried solving this with a retry mechanism in write without response, but half the time it has been an underlying problem in my firmware so I never definitively solved the problem.